tests/simulator/test_advance.py

```diff
@@ -237,6 +240,106 @@ def test_field_coarsening_via_subcycles(self, refinement_boxes):
+        coarsest_time_after = coarsest_time_before + lvlSteps[0]
+        assert_time_in_hier(coarsest_time_before, coarsest_time_after)
+
+        coarseBeforePatches = L0_datahier.level(0, coarsest_time_before).patches
+        coarseAfterPatches = L0_datahier.level(0, coarsest_time_after).patches
+        assert len(coarseBeforePatches) == len(coarseAfterPatches)
+
+        for coarsePatch_idx in range(len(coarseBeforePatches)):
+            coarseBeforePatch = coarseBeforePatches[coarsePatch_idx]
+            coarseAfterPatch  = coarseAfterPatches[coarsePatch_idx]
+            assert coarseBeforePatch.box == coarseAfterPatch.box
+
+            for finer_subcycle in range(0, global_vars.sim.level_step_nbr[1] + 1):
+
+                finer_subcycle_time = coarsest_time_before + (lvlSteps[1] * finer_subcycle)
+                assert_time_in_hier(finer_subcycle_time)
+                finer_subcycle_patches = L0L1_datahier.level(1, finer_subcycle_time).patches
+
+                for finer_subcycle_patch in finer_subcycle_patches:
+                    if boxm.coarsen(finer_subcycle_patch.box, refinement_ratio) in coarseBeforePatch.box:
+                        for EM in ["E", "B"]:
+                            for xyz in ["x", "y", "z"]:
+                                qty = f"{EM}{xyz}"
+                                assert qty in coarseBeforePatch.patch_datas
+
+                                coarseBefore_pd = coarseBeforePatch.patch_datas[qty]
+                                refinedInterpolatedField = refine(
+                                  coarseBefore_pd,
+                                  data=interpolate_datasets(
+                                    coarsest_time_before, coarsest_time_after, finer_subcycle_time,
+                                    coarseBefore_pd.dataset[:], coarseAfterPatch.patch_datas[qty].dataset[:])
+                                )
+
+                                finer_subcycle_pd = finer_subcycle_patch.patch_datas[qty]
+                                ghosts_nbr = finer_subcycle_pd.ghosts_nbr
+
+                                for finer_ghost_box in finer_subcycle_pd.ghost_box - finer_subcycle_patch.box:
+                                    is_upper = np.array(finer_ghost_box.lower > finer_subcycle_patch.box.lower, dtype=int)
+                                    finer_ghost_box = boxm.shift(finer_ghost_box, is_upper and coarseBefore_pd.primal_directions())
+                                    lvlOverlap = refinedInterpolatedField.box * finer_ghost_box
+
+                                    if lvlOverlap is not None:
+                                        finer_ghostbox_data = finer_subcycle_pd[finer_ghost_box]
+                                        refinedInterpGhostBox_data = refinedInterpolatedField[boxm.shift(finer_ghost_box, ghosts_nbr)]
+                                        np.testing.assert_allclose(finer_ghostbox_data, refinedInterpGhostBox_data, atol=1e-6)
```

Comment:
so, when I see this 1e-6 I think oh ok that's because we dump in floats... but really in this kind of test we expect binary equality (I think?) so shouldn't we check for that?